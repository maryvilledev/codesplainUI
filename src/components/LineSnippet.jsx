import React, { PropTypes } from 'react';
import CodeMirror from 'react-codemirror';

// Base options for CodeMirror instances for a LineSnippet
const baseOptions = {
  lineNumbers: true,
  theme: 'codesplain',
  readOnly: true,
  cursorBlinkRate: -1,
  lineWrapping: true,
};

class LineSnippet extends React.Component {
  componentDidMount() {
    const cm = this.codeMirror.getCodeMirror();
    cm.setSize('auto', 'auto');
  }
  render() {
    const {
      lineNumber,
      value,
    } = this.props;
    const codeMirrorOptions = {
      ...baseOptions,
      firstLineNumber: lineNumber,
    };
    return (
      <CodeMirror
        ref={(cm) => { this.codeMirror = cm; }}
        value={value}
        options={codeMirrorOptions}
      />
    );
  }
}

LineSnippet.propTypes = {
  lineNumber: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired,
};

export default LineSnippet;
