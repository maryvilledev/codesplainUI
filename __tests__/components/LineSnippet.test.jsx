import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import LineSnippet from '../../src/components/LineSnippet';

describe('<LanguageSelector />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <LineSnippet
        lineNumber={1}
        value={'foo bar baz'}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('prop: value', () => {
    it('is passed to the CodeMirror instance', () => {
      const annotation = 'Wubba lubba dub dub!';
      const wrapper = shallowWithContext(
        <LineSnippet
          lineNumber={1}
          value={annotation}
        />,
      );
      expect(wrapper.find('CodeMirror').prop('value')).toEqual(annotation);
    });
  });
  describe('prop: lineNumber', () => {
    it('is passed to the firstLineNumber option', () => {
      const lineNumber = 0;
      const annotation = 'Wubba lubba dub dub!';
      const wrapper = shallowWithContext(
        <LineSnippet
          lineNumber={lineNumber}
          value={annotation}
        />,
      );
      expect(wrapper.find('CodeMirror').prop('options').firstLineNumber).toEqual(lineNumber);
    });
  });
});
