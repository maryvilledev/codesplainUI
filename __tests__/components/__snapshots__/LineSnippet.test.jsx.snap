exports[`<LanguageSelector /> matches snapshot 1`] = `
<CodeMirror
  options={
    Object {
      "cursorBlinkRate": -1,
      "firstLineNumber": 1,
      "lineNumbers": true,
      "readOnly": true,
      "theme": "codesplain",
    }
  }
  preserveScrollPosition={false}
  value="foo bar baz" />
`;
